packages/flutter/lib/src/material/chip.dart

```diff
@@ -1830,25 +1819,36 @@ class _RenderChip extends RenderBox with SlottedContainerRenderObjectMixin<_Chip
     return _computeSizes(constraints, ChildLayoutHelper.dryLayoutChild).size;
   }
 
+  @override
+  double? computeDryBaseline(BoxConstraints constraints, TextBaseline baseline) {
+    final _ChipSizes sizes = _computeSizes(constraints, ChildLayoutHelper.dryLayoutChild);
+    final BaselineOffset labelBaseline = BaselineOffset(label.getDryBaseline(sizes.labelConstraints, baseline))
+        + (sizes.content - sizes.label.height + sizes.densityAdjustment.dy) / 2
+        + theme.padding.top + theme.labelPadding.top;
+    return labelBaseline.offset;
+  }
+
   _ChipSizes _computeSizes(BoxConstraints constraints, ChildLayouter layoutChild) {
     final BoxConstraints contentConstraints = constraints.loosen();
     // Find out the height of the label within the constraints.
-    final Offset densityAdjustment = Offset(0.0, theme.visualDensity.baseSizeAdjustment.dy / 2.0);
-    final Size rawLabelSize = layoutChild(label, contentConstraints);
+    final Size rawLabelSize = label.getDryLayout(contentConstraints);
```

Comment:
The label will be laid out a second time in this method, with a set of different constraints. Using `getDryLayout` prevent the child from actually changing size and creating other side effects. The downside of using `getDryLayout` is that you won't be able to use `LayoutBuilder`